src/sbt-test/sbt-autoversion/simple/test

```diff
@@ -0,0 +1,104 @@
+# Generate project files.
+> reload
+> compile
+
+# Setup repository.
+$ exec git init
+$ exec git add .
+$ exec git commit -m 'initial commit'
+$ exec git tag v0.0.0
+
+# Prove sbt can start.
+> reload
+> compile
+> show version
+
+# 0.0.0 => 0.0.1 => 0.0.2-SNAPSHOT
+$ exec git commit --allow-empty -m 'fix: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.0.2-SNAPSHOT"' version.sbt
+
+# 0.0.2-SNAPSHOT => 0.0.2 => 0.0.3-SNAPSHOT
+$ exec git commit --allow-empty -m 'bugfix: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.0.3-SNAPSHOT"' version.sbt
+
+# 0.0.3-SNAPSHOT => 0.0.3 => 0.0.4-SNAPSHOT
+$ exec git commit --allow-empty -m 'patch: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.0.4-SNAPSHOT"' version.sbt
+
+# 0.0.4-SNAPSHOT => 0.1.0 => 0.1.1-SNAPSHOT
+$ exec git commit --allow-empty -m 'feature: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.1.1-SNAPSHOT"' version.sbt
+
+# 0.1.1-SNAPSHOT => 0.1.1 => 0.1.2-SNAPSHOT
+$ exec git commit --allow-empty -m '[fix]: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.1.2-SNAPSHOT"' version.sbt
+
+# 0.1.2-SNAPSHOT => 0.1.2 => 0.1.3-SNAPSHOT
+$ exec git commit --allow-empty -m '[bugfix]: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.1.3-SNAPSHOT"' version.sbt
+
+# 0.1.3-SNAPSHOT => 0.2.0 => 0.2.1-SNAPSHOT
+$ exec git commit --allow-empty -m '[minor]: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.2.1-SNAPSHOT"' version.sbt
+
+# 0.2.1-SNAPSHOT => 0.3.0 => 0.3.1-SNAPSHOT
+$ exec git commit --allow-empty -m '[feature]: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"0.3.1-SNAPSHOT"' version.sbt
+
+# 0.3.1-SNAPSHOT => 1.0.0 => 1.0.1-SNAPSHOT
+$ exec git commit --allow-empty -m '[major]: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"1.0.1-SNAPSHOT"' version.sbt
+
+# 1.0.1-SNAPSHOT => 1.0.1 => 1.0.2-SNAPSHOT
+$ exec git commit --allow-empty -m '[fix]: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"1.0.2-SNAPSHOT"' version.sbt
+
+# 1.0.2-SNAPSHOT => 1.0.2 => 1.0.3-SNAPSHOT
+$ exec git commit --allow-empty -m 'patch: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"1.0.3-SNAPSHOT"' version.sbt
+
+# 1.0.3-SNAPSHOT => 1.1.0 => 1.1.1-SNAPSHOT
+$ exec git commit --allow-empty -m 'feature: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"1.1.1-SNAPSHOT"' version.sbt
+
+# 1.1.1-SNAPSHOT => 1.2.0 => 1.2.1-SNAPSHOT
+$ exec git commit --allow-empty -m 'minor: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"1.2.1-SNAPSHOT"' version.sbt
+
+# 1.2.1-SNAPSHOT => 2.0.0 => 2.0.1-SNAPSHOT
+$ exec git commit --allow-empty -m '[breaking]: foobar'
+> reload
+> release with-defaults
+$ exec grep -Fq '"2.0.1-SNAPSHOT"' version.sbt
+
+# 2.0.1-SNAPSHOT => 2.0.1 => 2.0.2-SNAPSHOT
+$ exec git commit --allow-empty -m 'doing hard work'
```

Comment:
I also added an example of an unconventional commit, i.e. one that exercises the catch-all.